src/plugins/auto/plugin.cpp

```diff
@@ -114,10 +114,25 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
 
         return "";
     };
-
+    auto checkPriorityConfig = [&] (const std::string& priString) {
+        std::string::size_type pos = 0;
+        std::string::size_type endpos = 0;
+        while ((endpos = priString.find(",", pos)) != std::string::npos) {
+            auto subStr = priString.substr(pos, endpos - pos);
+            if (subStr.find("-") != 0)
```

Comment:
do you mean (subStr.find("-") != std::string::npos) here,  would "AUTO:-***" enable DevicePriority  ?

Reply:
AUTO:-*** will not enable devicepriority.
this piece of logic is to traverse the priority string, and if all the device start with "-", then disable priority, if any of device is not started with "-" (means it's a real device in priority list), then enable device priority.